playbooks/common/openshift-master/scaleup.yml

```diff
@@ -0,0 +1,42 @@
+---
+- include: ../openshift-cluster/evaluate_groups.yml
+
+- name: Gather facts
+  hosts: oo_etcd_to_config:oo_masters_to_config:oo_nodes_to_config
+  roles:
+  - openshift_facts
+
+- name: Update master count
+  hosts: oo_masters:!oo_masters_to_config
+  serial: 1
+  vars:
+    openshift_master_count: "{{ groups.oo_masters | length }}"
+  roles:
+  - openshift_facts
+  post_tasks:
+  - name: Update master count
+    lineinfile:
+      dest: "{{ openshift.common.config_base}}/master/master-config.yaml"
+      regexp: '^  masterCount:.*$'
+      line: "  masterCount: {{ openshift_master_count }}"
+    register: master_config
+  - name: Restart master api
+    service: name={{ openshift.common.service_type }}-master-api state=restarted
+    when: master_config | changed
```

Comment:
Also, having the service restarts as handlers might be a better than having them in post_tasks as well.